Rakefile

```diff
@@ -1,7 +1,20 @@
-if ENV['RACK_ENV'] != 'production'
-  require 'rspec/core/rake_task'
-  
-  RSpec::Core::RakeTask.new :spec
-  
-  task default: [:spec]
+require 'data_mapper'
+require './app/app.rb'
+require 'rspec/core/rake_task'
+RSpec::Core::RakeTask.new :spec
+
+task default: [:spec]
+
+namespace :db do
+  desc "Non destructive upgrade"
+  task :auto_upgrade do
+    DataMapper.auto_upgrade!
+    puts "Auto-upgrade complete (no data loss)"
+  end
+
+  desc "Destructive upgrade"
+  task :auto_migrate do
+    DataMapper.auto_migrate!
+    puts "Auto-migrate complete (data was lost)"
+  end
 end
```

Comment:
I love that you're leveraging rake to simplify common tasks and separate them out to their own 'tool-space'.